parsers/iceland.py

```diff
@@ -0,0 +1,93 @@
+import sys
+import requests
+import json
+import csv
+import io
+
+from collections import defaultdict
+from datetime import datetime, timedelta
+from .utils import store_data
+
+# ------------------------------------------------------------------------
+# Globals
+
+URL_CASES_CUM = "https://raw.githubusercontent.com/tryggvigy/CoronaWatchIS/master/data/covid_in_is.cvs"
+URL_HOSPITALIZED_CUM = "https://raw.githubusercontent.com/tryggvigy/CoronaWatchIS/master/data/covid_in_is_hosp.cvs"
+
+LOC = "case-counts/Europe/Western Europe/Iceland"
+cols = ['time', 'cases', 'deaths', 'hospitalized', 'ICU', 'recovered']
+
+# ------------------------------------------------------------------------
+# Functions
+
+def parse_date(s):
+    try:
+      return datetime.strftime(datetime.strptime(s,'%m-%d-%Y'),'%Y-%m-%d')
+    except ValueError:
+        return s
+
+def sorted_date(s):
+    return sorted(s, key=lambda d: datetime.strptime(d[cols.index('time')], "%Y-%m-%d"))
+
+
+# ------------------------------------------------------------------------
+# Sub parsers
+
+
+def parse_csv(regions_date, url, col_index, output_column):
+    r = requests.get(url)
+    if not r.ok:
+        print(f"Failed to fetch {url}", file=sys.stderr)
+        exit(1)
+        r.close()
+
+    fd = io.StringIO(r.text)
+    rdr = csv.reader(fd)
+    next(rdr)
+
+    for row in rdr:
+        date_string = parse_date(row[0])
+        regions_date["Iceland"][date_string][output_column] = row[col_index]
+
+
+def parse_cases(regions_date):
+    parse_csv(regions_date, URL_CASES_CUM, 4, 'cases')
+
+def parse_deaths(regions_date):
+    parse_csv(regions_date, URL_CASES_CUM, 7, 'deaths')
+
+def parse_hospitalized(regions_date):
+    parse_csv(regions_date, URL_HOSPITALIZED_CUM, 4, 'hospitalized')
+
+def parse_icu(regions_date):
+    parse_csv(regions_date, URL_HOSPITALIZED_CUM, 2, 'hospitalized')
```

Comment:
This should be ", 'ICU')" instead of ", 'hospitalized')", right?

Reply:
Thanks @noleti! Can't believe I missed this. Stayed up quite late yesterday to finish this up so I blame tiredness :)